src/Tribe/Aggregator/Record/Batch_Queue.php

```diff
@@ -0,0 +1,422 @@
+	}
+
+	/**
+	 * After the process has been completed make sure the `post_modified` and `post_status` are updated accordingly.
+	 *
+	 * @since TBD
+	 *
+	 * @return $this
+	 */
+	protected function complete() {
+		try {
+			// Updates the Modified time for the Record Log.
+			$args = [
+				'ID'            => $this->record->post->ID,
+				'post_modified' => ( new DateTime( 'now', new DateTimeZone( 'UTC' ) ) )->format( Tribe__Date_Utils::DBDATETIMEFORMAT ),
+				'post_status'   => Tribe__Events__Aggregator__Records::$status->success,
+			];
+			wp_update_post( $args );
+		} catch ( Exception $e ) {
+			do_action( 'tribe_log', 'debug', __METHOD__, [
+				'message'   => $e->getMessage(),
+				'record'    => $this->record,
+				'exception' => $e,
+			] );
+		}
+
+		return $this;
+	}
+
+	/**
+	 * Processes a batch for the queue
+	 *
+	 * @since TBD
+	 *
+	 * @throws Exception
+	 *
+	 * @param null $batch_size The batch size is ignored on batch import as is controlled via the initial filtered value.
+	 *
+	 * @return self|Tribe__Events__Aggregator__Record__Activity
+	 */
+	public function process( $batch_size = null ) {
+		// This batch has not started yet, make sure to initiate this import.
+		if ( empty( $this->record->meta['batch_started'] ) ) {
+			$now = new DateTime( 'now', new DateTimeZone( 'UTC' ) );
+			$this->record->update_meta(
+				'batch_started',
+				$now->format( Tribe__Date_Utils::DBDATETIMEFORMAT )
+			);
+			$this->record->update_meta( Tribe__Events__Aggregator__Record__Queue::$queue_key, 'fetch' );
+			$this->record->set_status_as_pending();
+			$this->start();
+
+			return $this;
+		}
+
+		if ( $this->record->post instanceof WP_Post && ! $this->is_in_progress() ) {
+			return $this;
+		}
+
+		return $this->activity();
+	}
+
+	/**
+	 * Create the initial request to the EA server requesting that the client is ready to start getting batches of events.
+	 *
+	 * @since TBD
+	 */
+	public function start() {
+		if (
+			empty( $this->record->meta['allow_batch_push'] )
+			|| empty( $this->record->meta['import_id'] )
+			|| empty( $this->record->meta['next_batch_hash'] )
+		) {
+			$error = new WP_Error(
+				'core:aggregator:invalid-batch-record',
+				esc_html__( 'The batch registered does not have all the required fields.', 'the-events-calendar' )
+			);
+			$this->record->set_status_as_failed( $error );
+
+			return;
+		}
+
+		/** @var Tribe__Events__Aggregator__Service $service */
+		$service = tribe( 'events-aggregator.service' );
+
+		if ( $service->is_over_limit( true ) ) {
+			$this->record->update_meta( 'last_import_status', 'error:usage-limit-exceeded' );
+			$this->record->set_status_as_failed();
+
+			return;
+		}
+
+		$version                 = $service->api['version'];
+		$service->api['version'] = 'v2.0.0';
+
+		$body = [
+			'batch_size'       => apply_filters( 'event_aggregator_event_batch_size', 10 ),
+			'batch_interval'   => apply_filters( 'event_aggregator_event_batch_interval', 10 ),
```

Comment:
Needs doc blocks, and likely needs to be on it's own method.

Reply:
Great point, about the method tho I'm not sure the difference or benefit in this instance, would you mind clarify a bit more, thanks

Reply:
Usually we dedicate a method to the filtering, e.g. `$batch_size = $this->get_batch_size()` and move the filtering there.